features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/resource/RequestExt.kt

```diff
@@ -0,0 +1,52 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.rum.resource
+
+import com.datadog.android.api.instrumentation.network.RequestInfo
+import com.datadog.android.lint.InternalApi
+import java.io.IOException
+import java.util.UUID
+
+/**
+ * Generates an identifier to uniquely track requests.
+ */
+@InternalApi
+fun RequestInfo.buildResourceId(generateUuid: Boolean): ResourceId {
```

Comment:
I think we should avoid having it as extension method, because it will be automatically shown in the suggestions dropdown for `RequestInfo` which is a public API, not internal.